components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/ValidateCodeApiServiceImpl.java

```diff
@@ -28,8 +30,18 @@ public Response validateCodePost(CodeValidationRequestDTO codeValidationRequestD
         try {
             NotificationPasswordRecoveryManager notificationPasswordRecoveryManager = RecoveryUtil
                     .getNotificationBasedPwdRecoveryManager();
-            user = notificationPasswordRecoveryManager
-                    .getValidatedUser(codeValidationRequestDTO.getCode(), codeValidationRequestDTO.getStep());
+            String code = codeValidationRequestDTO.getCode();
+            try {
+                String hashedCode = Utils.doHash(code);
+                user = notificationPasswordRecoveryManager
+                        .getValidatedUser(hashedCode, codeValidationRequestDTO.getStep());
```

Comment:
Are we sure that all the places that call this method pass the hashed code? Wouldn't it easy to handle this inside the `getValidatedUser` method? It would make it easy to handle backward compatibility as well.

Reply:
Move to `getValidatedUser` method with 7d6d0fd7b9a5e2d6e08e1be01b492b377b472666